projects/ids-enterprise-ng/src/lib/modal-dialog/soho-modal-dialog.ref.ts

```diff
@@ -426,6 +428,10 @@ export class SohoModalDialogRef<T> {
    * @return true if the dialog can be closed; otherwise false if veoted.
    */
   private onBeforeClose(event: any): boolean {
+    this.ngZone.runOutsideAngular(() => {
+      this.componentContainer.appendChild(this.componentRef.location.nativeElement);
+    });
```

Comment:
Surprisingly, if you try to move this just below the `appendChild()` change in `onBeforeOpen()` method above, it will not leak ExampleModalDialogComponent. But of course you won't see the component inside the modal. 

I guess angular component's references won't be broken as long as you it didn't undergo its lifecycle while its outside it's original place in DOM.

Reply:
The issue here (at least when I implemented this) was that the enterprise code reparented the dialog which meant angular lost track of it.  It should be the case that calling close destroyed all the markup - looks like there are still issues with this.

Reply:
Would it be better to get the standard enterprise code to keep the dialog in the defined location?

Reply:
Aggree! Can we put in some sort of an issue breaking this down in EP?

Reply:
Update: I was able to get around moving the component in the DOM by doing `AppRef.attachView(childComponentRef.hostView);` because that method won't attach the component to a parent container so doing an appendChild to the modal is okay. 

However the next thing that's blocking me is when I call the method to destroy the component and modal inside the button click callback, the component still doesn't get GC'd. But if I call the method that destroys the component outside that button click callback, it's being GC'd properly..

Reply:
In the click handler try wrapping the code in angular zone: 

```typescript
this.ngZone.run(() => { ... });
```

Reply:
unfortunately still no luck with that

Reply:
yup I tested that area too a lot of times actually, so I ruled out that one out since the only remaining issue is with ExampleModalDialogComponent references.

Reply:
@tmcconechy  - did we check if EP 'moving' the dialog's markup was causing the issue here?

Reply:
@bthharper I did not check that? Which issue is it?

Reply:
I remember (it was a while ago) that when the dialog was displayed (opened) EP would move (or copy) the markup to a placeholder before displaying it.   This did cause issues destroying the markup which we had to fix.  I missed the memory leaks, but I was wondering if angular was not getting told to remove the component - and any references?